trolldb/tests/tests_api/test_api.py

```diff
@@ -26,6 +30,44 @@ def document_ids_are_correct(test_ids: list[str], expected_ids: list[str]) -> bo
     return Counter(test_ids) == Counter(expected_ids)
 
 
+def single_query_is_correct(key: str, value: str | datetime) -> bool:
+    """Checks if the given single query, denoted by ``key`` matches correctly against the ``value``."""
+    return (
+            Counter(http_get(f"queries?{key}={value}").json()) ==
+            Counter(TestDatabase.match_query(**{key: value}))
+    )
+
+
+def query_results_are_correct(keys: list[str], values_list: list[list[str | datetime]]) -> bool:
+    """Checks if the retrieved result from querying the database via the API matches the expected result.
+
+    There can be more than one query `key/value` pair.
+
+    Args:
+        keys:
+            A list of all query keys, e.g. ``keys=["platform", "sensor"]``
+
+        values_list:
+            A list in which each element is a list of values itself. The `nth` element corresponds to the `nth` key in
+            the ``keys``.
+
+    Returns:
+        A boolean flag indicating whether the retrieved result matches the expected result.
+    """
+    # Make a single query string for all queries
+    query_buffer = []
+    for label, value_list in zip(keys, values_list, strict=True):
+        query_buffer += [f"{label}={value}" for value in value_list]
+    query_string = "&".join(query_buffer)
```

Comment:
Having a comment here really means this should be its own function, called eg `make_single_query_from_all_queries`

Reply:
Done!